src/components/AnalysisPage/3dModelViewer/setupThreeScene.js

```diff
@@ -47,86 +47,96 @@ const setupThreeScene = (canvas, objectURL, fileType) => {
 
   const disposables = [];
 
-  let loader;
-  if (fileType === "stl") {
-    loader = new STLLoader();
-  } else if (fileType === "ply") {
-    loader = new PLYLoader();
-  } else {
-    throw new Error(`Unsupported file type`);
-  }
-
-  loader.load(objectURL, function (geometry) {
-    const materialOptions = {
-      specular: 0xffffff,
-      shininess: 100,
-      flatShading: true,
-    };
-
-    if (fileType === "ply") {
-      geometry.computeVertexNormals();
-      materialOptions.vertexColors = true;
+  const loadModel = (objectURL, fileType) => {
+    console.log("Loading model");
+    // clearModel();
+    let loader;
+    if (fileType === "stl") {
+      loader = new STLLoader();
+    } else if (fileType === "ply") {
+      loader = new PLYLoader();
+    } else {
+      throw new Error(`Unsupported file type`);
```

Comment:
The toggle works a lot better now but I think theres one more optimization we can make when switching between the grey model and the thickness visualization model.  Once we load in a ply file, instead of calling the loader every time to load back in the original file to get the grey colored view, we can just change the color of the model using threejs materials. So instead of loading in the original part file every time to get the grey view we can just change the material on the mesh to show grey.

This will solve the little pause between switching out the models too. Currently if you load in a really big file it can take 4-5 seconds to switch between the views when you press the toggle because it needs to load the whole part into the loader each time.

You've already mentioned wanting to fix the little flash of the part when switching the toggle so I think this might be a good way to go about it

Reply:
This should be good now too!